tests/integration/DatabaseUpdateTest.php

```diff
@@ -10,25 +10,85 @@
  */
 class DatabaseUpdateTest extends \PHPUnit_Framework_TestCase {
 
+	/**
+	 * @var \WMDE\Fundraising\Store\Factory
+	 */
+	private $factory;
+
+	/**
+	 * @var \Doctrine\DBAL\Schema\AbstractSchemaManager
+	 */
+	private $schemaManager;
+
+	public function setUp() {
+		$this->factory = TestEnvironment::newDefault()->getFactory();
+		$this->schemaManager = $this->factory->getConnection()->getSchemaManager();
+	}
+
 	public function testMissingTableIsThereAfterUpdate() {
-		$factory = TestEnvironment::newDefault()->getFactory();
-		$entityManager = $factory->getEntityManager();
+		$schemaTo = $this->schemaManager->createSchema();
+		$schemaTo->dropTable( 'action_log' );
 
-		$schemaTool = new SchemaTool( $entityManager );
-		$actionLogMetaData = $entityManager->getMetadataFactory()->getMetadataFor( 'WMDE\Fundraising\Entities\ActionLog' );
-		$schemaTool->dropSchema( array( $actionLogMetaData ) );
+		$this->updateDatabaseBySchema( $schemaTo );
 
 		$this->assertNotContains(
 			'public.action_log',
-			$factory->getConnection()->getSchemaManager()->createSchema()->getTableNames()
+			$this->schemaManager->createSchema()->getTableNames()
 		);
 
-		$factory->newUpdater()->update();
+		$this->factory->newUpdater()->update();
 
 		$this->assertContains(
 			'public.action_log',
-			$factory->getConnection()->getSchemaManager()->createSchema()->getTableNames()
+			$this->schemaManager->createSchema()->getTableNames()
+		);
+	}
+
+	public function testMissingColumnIsThereAfterUpdate() {
+		$schemaTo = $this->schemaManager->createSchema();
+		$schemaTo->getTable( 'action_log' )->dropColumn( 'al_username' );
+
+		$this->updateDatabaseBySchema( $schemaTo );
+
+		$this->assertArrayNotHasKey(
```

Comment:
This violates Arrange Act Assert. As does the change I made to the already existing test a few days ago. In this case I think it makes sense to break the rule, though do keep it in mind for other things. Just saying - this commit is perfectly fine - I want to avoid starting a series of actual bad AAA violations :)